### ISSUE TITLE
[FEATURE] Pass layer data into cpgv.init as parameter

### ISSUE THREAD

**Issue by G-Boyce** [Canadian-Geospatial-Platform/geoview#2411]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Who request this Feature

Greg Boyce, Data Management Plan tool, NRCan

### User Story

I'd like to be able to pass GeoJSON data (e.g. as a text string) into the cgpv.init function.

### Implementation UI

Rather than being forced to read GeoJSON data from a fixed file on the web server, I'd like to dynamically build up the data using an SQL query in PHP, and pass that GeoJSON string into the cgpv.init function.  (Instead of parsing the file, parse the passed string.)

Otherwise, I'll likely need to create files on the web server, which gets messy (disk space checks, regular file cleanup, etc).

Note:  we already have a web page which creates HTML/JavaScript output from SQL queries, so setting up a new WFS server to also query this data, would presumably double the database query load.

Data passed into cgpv.init, should probably be **_in addition to_** layers already defined in the map Config section.  (E.g. you have several layers well defined by WFS or other layer types in the map config section, and just want to add data for one dynamically constructed layer.)

So, using the following Config string (from [GeoJSON layer page](https://canadian-geospatial-platform.github.io/geoview/public/geojson.html)):
```
        "metadataAccessPath": {
          "en": "./datasets/geojson/metadata.json"
        },
        "geoviewLayerType": "GeoJSON",
        "listOfLayerEntryConfig": [
          {
            "layerId": "polygons.json",
            "layerName": {
              "en": "Polygons"
            },
            "layerFilter": "creationDate >= date '2020/02/15'"
          },
```

Basically allowing the contents of "polygons.json" to be passed into cgpv.init as a parameter.

The metadataAccessPath (metadata.json) could remain in the config (or be passed in, too).

**Comment by jolevesq** [comment]
Test

**Comment by jolevesq** [comment]
@G-Boyce Mention

**Comment by fletour** [comment]
To summarise our feature request. @G-Boyce, please edit if something is missing.

- Load dynamically a layer (for now, in geoJSON) into geoview, including the data attributes. 
- Each time there is a change, the full layer is reloaded into the map, no need to remove or add a feature programmatically
- The properties associated with the data layer (attributes) should be visible in a pop-up over the map ou beside in a panel.
- Symbolisation is done in a static file defined internally in the system (no dynamic symbology change)

**Comment by jolevesq** [comment]
@G-Boyce The code is now ready to test: https://canadian-geospatial-platform.github.io/geoview/public/demo-geojson-inject.html

**Comment by G-Boyce** [comment]
Seems to be working so far.  Thanks!
